utils/filesystem.go

```diff
@@ -104,3 +111,81 @@ func PingHealth(healthEndpoint string) bool {
 	}
 	return started
 }
+
+//GetZipURL from github api /repos/:owner/:repo/:archive_format/:ref
+func GetZipURL(c *cli.Context) string {
+	branch := c.String("branch")
+	owner := c.String("owner")
+	repo := c.String("repo")
+
+	client := github.NewClient(nil)
+
+	opt := &github.RepositoryContentGetOptions{Ref: branch}
+
+	URL, _, err := client.Repositories.GetArchiveLink(context.Background(), owner, repo, "zipball", opt)
+	if err != nil {
+		log.Fatal(err)
+	}
+	url := URL.String()
+	fmt.Println(URL)
+	return url
+}
+
+//DownloadFile - the git zip of master into current dir
+func DownloadFile(zipFileName, url string) error {
+
+	// Get the data
+	resp, err := http.Get(url)
+	errors.CheckErr(err, 400, "")
+	defer resp.Body.Close()
+
+	// Create the file
+	out, err := os.Create(zipFileName)
+	errors.CheckErr(err, 401, "")
+	defer out.Close()
+
+	// Write body to file
+	_, err = io.Copy(out, resp.Body)
+	fmt.Println(zipFileName)
+
+	return err
+}
+
+//UnZip downloaded file
+func UnZip(zipFileName, destination string) {
+	zipReader, _ := zip.OpenReader(zipFileName)
+
```

Comment:
Also, would you say this function is doing more than unzipping? Is it copying a zipped file to the file system  destination? Maybe the function name should reflect that?

Reply:
No, this is just unzipping the contents of the zip file into the destination specified by the user. The range loop is iterating over each element within the zip file and putting it into the correct directory. As attempted before, unzipping into the temp directory provides more issues when it comes to moving the files on the file system. (see previous commits)

Reply:
Yeah sorry, my comment was badly written. I'd still say this isn't an easy function to understand. Maybe something like `Unzip ToDestination(...)`. Not sure though